Docs: git conventions amended. (#9185)
ckeditor/ckeditor5: docs/framework/guides/contributing/git-commit-message-convention.md

Each commit can contain additional notes that will be inserted into the changelog:

* `MAJOR BREAKING CHANGE`,
* `MINOR BREAKING CHANGE`.

If any change contains the `MAJOR BREAKING CHANGE` note, the next release will be marked as `major` automatically.

For reference on how to identify minor or major breaking change see the {@link framework/guides/support/versioning-policy versioning policy guide}.

> Each `MAJOR BREAKING CHANGE` or `MINOR BREAKING CHANGE` note must be followed by the package name.

Reviewer: I think we could remove this rule. Any major BC causes the entire project to have a major release - hence, it's ok if we don't specify which package is affected (e.g. in case of DLLs). Same for minor - the entire project is bumped to the same version.

cc @pomek Do we need to change anything in the changelog generator?

Reviewer 2: https://github.com/ckeditor/ckeditor5-dev/blob/f60d89c17bbc3ab358ce5a103e5851ff68285f4c/packages/ckeditor5-dev-env/lib/release-tools/tasks/generatechangelogformonorepository.js#L483-L562

Reviewer 3: So what's the status for now? We leave the rule and the docs as-is? Or do we have any follow-up task for removing the rule?

<info-box>

> 	Remember to always specify whether the breaking change is `major` or `minor`. If you fail to do so, the system will assume all unspecified breaking changes are `major`.

Author: Fixed in commit.

</info-box>

### Package name

Most commits are related to one or more packages. Each affected package should be listed in parenthesis following the commit type. A package that was the most impacted by the change should be listed first.

It is, however, possible to skip this part if many packages are affected. This usually indicates a generic change. In this case having all the packages listed would reduce the changelog readability.

The package name is based on the npm package name, however, it has the `@ckeditor/ckeditor5-` prefix stripped.

If your change is related to the main package only, use `ckeditor5` as the package name.

<info-box>
	If the commit introduces a breaking change across the entire project (a generic change), the package name does not have to be specified.
</info-box>

### Entries ordering

The proper order for sections is as follows:

> * Public

Reviewer 2: Done.

> * Public
> * Internal
> * MAJOR/MINOR BREAKING CHANGES

Reviewer 2: ```suggestion
* Entries that should be added to the changelog
* Entries that will not be added to the changelog
* Breaking changes notes
```

> All entries must be separated with `\n`, otherwise the lines will not be treated separately.

Reviewer: What does it mean? New lines are always separated with `\n`. Otherwise they would not be new lines. So, I suppose you meant here "blank lines"?

Reviewer: OK, I've just noticed there's an example below. But anyway - "blank line" is a better term here.

Author: Fixed in commit.

### Examples of proper and improper message formatting

Example of a proper commit message:

```

> Type: Message 1.
> 
> Type: Message 2.
> ```

Reviewer 2: I agree, `Type` can be confusing.

```suggestion
Feature (scope-1): Message 1.

Fix (scope-2): Message 2.

Tests: The change across the entire project.
```
```

An example of an invalid commit message with incorrectly separated lines (the second line will just be treated as a part of the first line):

```

> Type: Message 1.
> Type: Message 2.

Reviewer 2: ```suggestion
Feature (scope-1): Message 1.
Fix (scope-2): Message 2.
Tests: Message 3.
```

```

An example of invalid comment message with a wrong section order (the "internal" message will be treated as a part of the BREAKING CHANGE message):

```

> Public: Message 1.

Reviewer: "Public" is confusing here. There's no such change type.

Author: Resolved in commit.

> Public: Message 1.

Reviewer 2: ```suggestion
Feature: Message 1.
```

Most commits are related to one or more packages. Each affected package should be listed in parenthesis following the commit type. A package that was the most impacted by the change should be listed first.

It is, however, possible to skip this part if many packages are affected. This usually indicates a generic change. In this case having all the packages listed would reduce the changelog readability.

The package name is based on the npm package name, however, it has the `@ckeditor/ckeditor5-` prefix stripped.

If your change is related to the main package only, use `ckeditor5` as the package name.

<info-box>
	If the commit introduces a breaking change across the entire project (a generic change), the package name does not have to be specified.
</info-box>

### Order of entries

The proper order for sections of a commit message is as follows:
* Entries that should be added to the changelog
* Entries that will not be added to the changelog
* Breaking change notes

All entries must be separated with `\n` (a blank line), otherwise the lines will not be treated separately.

### Examples of proper and improper message formatting

Example of a proper commit message:

```

> Feature (scope-1): Message 1.

Reviewer 2: ```suggestion
Feature (package-name-1): Message 1.
```

> Feature (scope-1): Message 1.

Reviewer 3: Why do we use `scope-1` etc. here and `package-name-1` etc. below? This is confusing, is this the same thing?

Author: Resolved in commit.

Fix (scope-2): Message 2.

Tests: The change across the entire project.

> 

Reviewer 3: Formatting is broken now, this code block lacks closing \`\`\`.

Each commit can contain additional notes that will be inserted into the changelog:

* `MAJOR BREAKING CHANGE`,
* `MINOR BREAKING CHANGE`.

> If any change contains the `MAJOR BREAKING CHANGE` note, the next release will be marked as `major` automatically.

Reviewer 3: ```suggestion
If any change contains the `MAJOR BREAKING CHANGE` note, the next release will automatically be marked as `major`.
```

Each `MAJOR BREAKING CHANGE` or `MINOR BREAKING CHANGE` note must be followed by the package name.

<info-box>
	Remember to always specify whether the breaking change is major or minor. If you fail to do so, the system will assume all unspecified breaking changes are major.
</info-box>

### Package name

Most commits are related to one or more packages. Each affected package should be listed in parenthesis following the commit type. A package that was the most impacted by the change should be listed first.

It is, however, possible to skip this part if many packages are affected. This usually indicates a generic change. In this case having all the packages listed would reduce the changelog readability.

The package name is based on the npm package name, however, it has the `@ckeditor/ckeditor5-` prefix stripped.

If your change is related to the main package only, use `ckeditor5` as the package name.

<info-box>
	If the commit introduces a breaking change across the entire project (a generic change), the package name does not have to be specified.
</info-box>

### Order of entries

> The proper order for sections of a commit message is as follows:

Reviewer 3: ```suggestion
The proper order of sections for a commit message is as follows:
```

> * Entries that should be added to the changelog

Reviewer 3: ```suggestion
* Entries that should be added to the changelog.
```

> * Entries that will not be added to the changelog

Reviewer 3: ```suggestion
* Entries that will not be added to the changelog.
```

> * Breaking change notes

Reviewer 3: ```suggestion
* Breaking change notes.
```

Author: I don't get this one. No comma anywhere and a dot at the last list item?

> All entries must be separated with a blank line, otherwise the lines will not be treated separately.

Reviewer 3: ```suggestion
All entries must be separated with a blank line, otherwise the lines will not be treated as separate entries.
```

> All entries must be separated with `\n` (a blank line), otherwise the lines will not be treated separately.

Reviewer: ```suggestion
All entries must be separated with a blank line, otherwise the lines will not be treated separately.
```

Reviewer: `\n` is a "new line character" so writing that it's a blank line is confusing.

> ### Examples of proper and improper message formatting

Reviewer 3: ```suggestion
### Examples of correct and incorrect message formatting
```

Example of a proper commit message:

```
Feature (package-name-1): Message 1.

Fix (package-name-2): Message 2.

> Tests: The change across the entire project.

Reviewer 3: ```suggestion
Tests: A change across the entire project.
```

```

An example of an invalid commit message with incorrectly separated lines (the second line will just be treated as a part of the first line):

```
Feature (package-name-1): Message 1.
Fix (package-name-2): Message 2.
Tests: Message 3.
```

> An example of an invalid commit message with an incorrect section order (the "internal" message will be treated as a part of the BREAKING CHANGE message):

Reviewer 3: ```suggestion
An example of an invalid commit message with an incorrect section order (the "internal" message will be treated as a part of the breaking change message):
```
